elide-async/src/main/java/com/yahoo/elide/async/hooks/AsyncAPIHook.java

```diff
@@ -58,6 +60,10 @@ protected void validateOptions(AsyncAPI query, RequestScope requestScope) {
      */
     protected void executeHook(LifeCycleHookBinding.Operation operation, LifeCycleHookBinding.TransactionPhase phase,
             AsyncAPI query, RequestScope requestScope, Callable<AsyncAPIResult> queryWorker) {
+        if (query.getAsyncAfterSeconds() == null) {
+            query.setAsyncAfterSeconds(elideSettings.getMaxAsyncAfterSeconds());
+        }
+
```

Comment:
We need a much better explanation (add a lot of details) why the hook is written this way.  It appears wrong to the casual reader.